### ISSUE TITLE
Improve finding nodes

### ISSUE THREAD

**Issue by cpojer** [facebook/jscodeshift#26]
We are discussing different ways to improve finding nodes. For now, we'll attempt to build a simple parse API that will allow us to pass a JS pattern into find and uses the existing pattern matching to match things. Let's see how far we take this.

```js
j.find(pattern`var $ = require('merge');`)

// becomes
j.find(j.VariableDeclaration, {
  init: {
    type: 'Identifier',
    name: undefined
  }
});
```

Alternative approaches would be to use CSS selectors, because they are made to query trees. Another solution would be to extend babylon/acorn but that seems hard to maintain over time.

**Comment by hzoo** [comment]
Yeah, maybe some inspiration from something similar like http://www.graspjs.com/?

**Comment by fkling** [comment]
@hzoo: That's exactly what I had in mind. I came across this project last year but forgot its name and couldn't find it anymore :(
